src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/directionsquestionnaire/ClaimantDirectionsQuestionnaireContentProvider.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.cmc.claimstore.documents.content.directionsquestionnaire;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.claimstore.documents.ClaimContentProvider;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.ResponseRejection;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.DirectionsQuestionnaire;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.cmc.claimstore.utils.Formatting.formatDateTime;
+
+@Component
+public class ClaimantDirectionsQuestionnaireContentProvider {
+
+    private final ClaimContentProvider claimContentProvider;
+    private final HearingContentProvider hearingContentProvider;
+
+    public ClaimantDirectionsQuestionnaireContentProvider(ClaimContentProvider claimContentProvider,
+                                                          HearingContentProvider hearingContentProvider) {
+        this.claimContentProvider = claimContentProvider;
+        this.hearingContentProvider = hearingContentProvider;
+    }
+
+    public Map<String, Object> createContent(Claim claim) {
+        DirectionsQuestionnaire claimantDirectionsQuestionnaire = claim.getClaimantResponse()
+            .filter(ResponseRejection.class::isInstance)
+            .map(ResponseRejection.class::cast)
+            .map(ResponseRejection::getDirectionsQuestionnaire)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .orElseThrow(IllegalArgumentException::new);
+
+        ImmutableMap.Builder<String, Object> contentBuilder = new ImmutableMap.Builder<>();
+        contentBuilder.putAll(claimContentProvider.createContent(claim));
+        contentBuilder.put("hearingContent", hearingContentProvider.mapDirectionQuestionnaire
+            .apply(claimantDirectionsQuestionnaire));
```

Comment:
mapDirectionQuestionnaire.apply in tests is fine but in code a functional method should be used as the assignment target for a lambda expression or method reference.

Reply:
I dont understand what you are trying to say. The Function is declared as target for lambda. But `.apply` to be used as assignment target ?